graph.go

```diff
@@ -265,3 +265,25 @@ func appendSortedMap(m map[string]interface{}, mustBracket bool, b io.Writer) {
 		fmt.Fprint(b, ";")
 	}
 }
+
+// VisitNodes visits all nodes recursively
+func (g Graph) VisitNodes(callback func(node Node)) {
+	for _, node := range g.nodes {
+		callback(node)
+	}
+
+	for _, subGraph := range g.subgraphs {
+		subGraph.VisitNodes(callback)
+	}
+}
+
+// FindNodeById return node by id
+func (g Graph) FindNodeById(id string) (foundNode Node, found bool) {
+	g.VisitNodes(func(node Node) {
```

Comment:
why not have
```
	g.VisitNodes(func(node Node) {
		if node.id == id {
			return node, true
		}
	})
	return Node{}, false
```

Reply:
moin. I thought that the return in the inner anonymous function can only refer to this. And that you cannot exit the outer function from the inner. Therefore the workaround by declaring when the callback doesn't need to be called is the only option to proved an early-exit.